### PR TITLE
Switch to AirBnb ESLint style guide

### PR DESCRIPTION
**Pull request by pdehaan**
Bump to the latest version of **eslint-config-universal-search** ESLint config (to be in sync w/ [**universal-search-addon**](https://github.com/mozilla/universal-search-addon/pull/141)).

Went pretty smooth, and mostly just resolving this new error:

```
error  Unexpected space before function parentheses  space-before-function-paren
```

Basically AirBnb style guide prefers `constructor() {` instead of `constructor () {`.

### CODE REVIEW

File: src/views/search_suggestions/search_suggestions_index_view.js [hunk 1]
```diff
@@ -7,15 +7,15 @@ import searchSuggestionsAdapter from '../../adapters/search_suggestions_adapter'
...
-  initialize () {
+  initialize() {
     this.adapter = searchSuggestionsAdapter;
     this.listenTo(this.adapter.combinedSuggestions, 'reset', this.render);
   },
 
-  afterRender () {
+  afterRender() {
     this.removeSubviews();
     console.log('adapter', this.adapter);
```
Comment (pdehaan) [inline src/views/search_suggestions/search_suggestions_index_view.js]: Not sure if we want to kill this line. I think it's currently the only `warning` we get from ESLint:

```
/Users/pdehaan/dev/github/universal-search-content/src/views/search_suggestions/search_suggestions_index_view.js
  17:5  warning  Unexpected console statement  no-console
```

File: .eslintrc [hunk 2]
```diff
@@ -1,7 +1,4 @@
-extends: universal-search
-
-parser: babel-eslint
+extends: universal-search/content
 
 rules:
-  no-console: 0
-  no-var: 2
+  func-names: 0
```
Comment (pdehaan) [inline .eslintrc:4]: I disabled this rule, but there really only 4 warnings generated by it currently:

```
/Users/pdehaan/dev/github/universal-search-content/src/adapters/search_suggestions_adapter.js
  26:70  warning  Missing function expression name  func-names
  29:68  warning  Missing function expression name  func-names

/Users/pdehaan/dev/github/universal-search-content/src/views/base_view.js
  44:49  warning  Missing function expression name  func-names

/Users/pdehaan/dev/github/universal-search-content/src/views/index_view.js
  68:38  warning  Missing function expression name  func-names
```

File: src/adapters/search_suggestions_adapter.js [hunk 3]
```diff
@@ -23,10 +23,10 @@ const SearchSuggestionsAdapter = State.extend({
         this.engine = data.engine;
         this.searchTerm = data.results.term;
 
-        this.remoteSuggestions.reset(_c.collect(data.results.remote, function (t) {
+        this.remoteSuggestions.reset(_c.collect(data.results.remote, function(t) {
           return { term: t, type: 'remote' };
         }));
-        this.localSuggestions.reset(_c.collect(data.results.local, function (t) {
+        this.localSuggestions.reset(_c.collect(data.results.local, function(t) {
```
Comment (pdehaan) [inline src/adapters/search_suggestions_adapter.js:29]: These anon-functions throw warnings if we tweak the `func-names` ESLint rule:

```
/Users/pdehaan/dev/github/universal-search-content/src/adapters/search_suggestions_adapter.js
  26:70  warning  Missing function expression name  func-names
  29:68  warning  Missing function expression name  func-names
```

Me thinks we can either give these a function name, or convert them to fat arrows.

File: src/views/base_view.js [hunk 4]
```diff
@@ -13,35 +13,35 @@ export default AmpersandView.extend({
...
-  sliceCollection (collection, CollectionType, start, n) {
+  sliceCollection(collection, CollectionType, start, n) {
     if (!collection.length) {
       return new CollectionType();
     }
     if (n === 1) {
       return new CollectionType([collection.at(start)]);
     }
-    return new CollectionType(collection.filter(function (item, itemIndex) {
+    return new CollectionType(collection.filter(function(item, itemIndex) {
```
Comment (pdehaan) [inline src/views/base_view.js:44]:
```
/Users/pdehaan/dev/github/universal-search-content/src/views/base_view.js
  44:49  warning  Missing function expression name  func-names
```

Not sure if we want to fat arrow this, or something.

File: src/views/index_view.js [hunk 5]
```diff
@@ -53,19 +53,19 @@ export default BaseView.extend({
...
     }
   },
 
-  _selectNextItem (increment) {
+  _selectNextItem(increment) {
     const items = document.querySelectorAll('li');
     let newlySelected;
 
-    Array.prototype.some.call(items, function (item, i) {
+    Array.prototype.some.call(items, function(item, i) {
```
Comment (pdehaan) [inline src/views/index_view.js:79]:
```
/Users/pdehaan/dev/github/universal-search-content/src/views/index_view.js
  68:38  warning  Missing function expression name  func-names
```

**Comment by jaredhirsch** [comment]
I think these non-essential PRs probably won't get much, if any, attention until after Mozlando. We'll get to them eventually, though.

**Comment by pdehaan** [comment]
:+1: Non-essential is my middle name!

**Comment by jaredhirsch** [comment]
@pdehaan aight, let's rebase this pup and see how it looks

File: src/views/index_view.js [hunk 6]
```diff
@@ -113,20 +113,20 @@ export default BaseView.extend({
...
+  resetAdapters() {
+    const adapters = [
       activityResultsAdapter.results,
       recommendedAdapter.results,
       searchSuggestionsAdapter.localSuggestions,
       searchSuggestionsAdapter.remoteSuggestions,
       searchSuggestionsAdapter.combinedSuggestions
     ];
-    for (let adapter in adapters) {
+    for (const adapter in adapters) { // eslint-disable-line guard-for-in
```
Comment (pdehaan) [inline src/views/index_view.js:129]: This is gross and makes me feel gross. If I don't disable the `guard-for-in` ESLint rule here, I get the following error:

```
/Users/pdehaan/dev/github/universal-search-content/src/views/index_view.js
  129:5  error  The body of a for-in should be wrapped in an if statement to filter unwanted properties from the prototype  guard-for-in

✖ 1 problem (1 error, 0 warnings)
```

Not sure if we want to:
a) Ignore the `guard-for-in` rule above.
b) Wrap body if an `if` to filter unwanted things.
c) Convert to a `forEach()` loop.

File: src/views/little_machine/little_machine_index_view.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    this.removeSubviews();
+    const items = this.sliceCollection(this.littleMachine.results,
+                                     LittleMachineResults, 0, 1);
+    this.renderCollection(items, LittleMachineItemView,
+                          '.little-machine-results');
+
+    // if there are results then show otherwise hide
+    this.littleMachine.results.length ? this.show() : this.hide(); // eslint-disable-line no-unused-expressions
+  }
+});
```
Comment (pdehaan) [inline src/views/little_machine/little_machine_index_view.js]: This seems weird. Did this file get deleted and I somehow re-added it via a bad rebase?

Comment (chuckharmston) [inline src/views/little_machine/little_machine_index_view.js]: It was renamed: https://github.com/mozilla/universal-search-content/commit/6fcfb8cc5e068e75edb093e0a2d5ff32f57db1c0

File: src/views/index_view.js (hunk 6 above)
Comment (jaredhirsch) [inline src/views/index_view.js:129]: Actually, `adapters` is an array. There's no reason to use for-in to iterate it, we can just do `adapters.forEach`.

**Comment by jaredhirsch** [comment]
I'm not sure it makes sense to use different eslint rules for idea-town vs this iframe vs the search addon.

I'll file a followup bug to unify the linting approach across the testpilot projects, and look into how close we can get the addon code without introducing babel and the related build chain complexity.

**Pull request closed without merging**